### PR TITLE
Add hexamine and precursor recipes

### PR DESCRIPTION
**Pull request by starlightsys**
Makes it possible to synthesize hexamine from ammonia and formaldehyde.
Adds formaldehyde and methanol, as well as recipes to make them.
references #25239 

#### Summary
SUMMARY: Content "Adds formaldehyde and methanol as precursors for hexamine, and recipes to make them."

#### Purpose of change
As discussed in #25239, some explosives related recipes are very challenging to find components for. This change makes hexamine simpler to come by, by adding a recipe for hexamine as well as its precursors.

#### Describe the solution
Adds a recipe for hexamine using ammonia and formaldehyde, adds the chemical ```chem_formaldehyde``` and the fuel ```chem_methanol```, a recipe for formaldehyde using methanol, and two recipes for methanol: one by distilling denatured alcohol (described as containing methanol), and one by distilling splintered wood or 2x4s.

#### Describe alternatives you've considered
Using denatured alcohol as a stand-in for methanol; making methanol not usable as a fuel.

### CODE REVIEW

**Comment by Mecares** [comment]
MAybe we could add some jars with preserved animals to musems and similar item groups that disassemble into a 3L Jar Formaldehyd and meat?

**Comment by vegarbg** [comment]
Great idea @Mecares!

**Comment by vegarbg** [comment]
Is there a way to make the jar of formaldehyde preserved human brain disassemble into a jar containing formaldehyde and a brain? Currently you would need a second container to receive the formaldehyde when you disassemble the jar.

**Comment by nexusmrsep** [comment]
Methanol is similar in taste to ethanol, but toxic. One of the symptoms of that toxicity is loss of vision that can be permanent long term. https://en.m.wikipedia.org/wiki/Methanol_toxicity
So my idea is: allow drinking methanol, on the same assumption as bleach, ie. with a warning popup, and short term blindness effect and high intoxication on drinking, just for flavor reasons.

**Comment by Pupsi-Mupsi** [comment]
Please add a linebreak to line 34 data/json/items/fuel.json

`  },{ `

**Comment by Pupsi-Mupsi** [comment]
jenkins rebuild

**Comment by I-am-Erk** [comment]
I will definitely be using those chemicals later. Thanks.

File: data/json/recipes/recipe_medsandchemicals.json [hunk 1]
```diff
@@ -2385,6 +2385,101 @@
...
+]
+},
+{
+"type" : "recipe",
+"result": "chem_methanol",
+"byproducts": [ [ "chem_ethanol", 2 ] ],
+"category": "CC_CHEM",
+"subcategory": "CSC_CHEM_CHEMICALS",
+"skill_used": "cooking",
+"difficulty": 4,
```
Comment (I-am-Erk) [inline data/json/recipes/recipe_medsandchemicals.json]: Distilling methanol from denatured alcohol is a pretty straightforward process, you might want to drop this to 3.

File: data/json/recipes/recipe_medsandchemicals.json [hunk 2]
```diff
@@ -2385,6 +2385,101 @@
...
+"autolearn": true,
+"qualities":[
+{"id":"BOIL","level":2},
+{"id":"DISTILL","level":1}
+], "tools": [
+[ [ "surface_heat", 25, "LIST" ] ]
+],
+"components": [
+[
+[ "splinter", 40 ]
```
Comment (I-am-Erk) [inline data/json/recipes/recipe_medsandchemicals.json]: A splinter is 1lb so this is 40lb of wood. Maybe allow 12 two by fours as an alternative ingred?

Comment (vegarbg) [reply data/json/recipes/recipe_medsandchemicals.json]: Sure.

I chose that number very arbitrarily. Let me know if it should be changed.

Comment (vegarbg) [reply data/json/recipes/recipe_medsandchemicals.json]: Yep, that yields 250mL.

**Comment by Rivet-the-Zombie** [comment]
It would be really nice if these new chemicals could be incorporated into existing recipes whenever possible.

**Pull request merged**